frontend/src/metabase/parameters/utils/dashboard-options.ts

```diff
@@ -71,9 +84,15 @@ export function getDefaultOptionForParameterSection() {
         defaultOption = options.find(o => o.type === "date/all-options");
       }
 
+      if (!defaultOption) {
+        throw new Error(
+          "No default option found for parameter section: " + sectionId,
+        );
+      }
+
       return [sectionId, defaultOption];
     }),
   );
 
-  return map;
+  return map as Record<SectionId, ParameterMappingOptions>;
```

Comment:
I used `as` here because `Object.fromEntries` returns `<string, T>` and not `<SectionId, T>`

Reply:
Actually `Object.fromEntries` returns `Partial<Record<SectionId, ParameterMappingOptions>>`. TS can't ensure that every `SectionId` key will be present.

Reply:
My point was that if you remove the cast and explicitly specify that this function returns `Partial<Record<SectionId, ParameterMappingOptions>>` - TS won't complain (in this file at least).